tests/functional/prometheous/test_prometheus_validations_file.py

```diff
@@ -333,3 +334,69 @@ def test_prometheus_pv_resize(self):
             metric_data=resize_metrics, operation="create")
+
+    @pytest.mark.tier3
+    def test_prometheus_volume_metrics_on_node_reboot(self):
+        """Validate volume metrics using prometheus before and after node
+        reboot"""
+
+        # Create PVC and wait for it to be in 'Bound' state
+        pvc_name = self.create_and_wait_for_pvc()
+        pod_name = openshift_ops.oc_create_tiny_pod_with_volume(
+            self._master, pvc_name, "autotest-volume",
+            image=self.io_container_image_cirros)
+        self.addCleanup(openshift_ops.oc_delete, self._master, 'pod', pod_name,
+                        raise_on_absence=False)
+        openshift_ops.wait_for_pod_be_ready(
+            self._master, pod_name, timeout=60, wait_step=2)
+
+        # Write data on the volume and wait for 2 mins and sleep is must for
+        # prometheus to get the exact values of the metrics
+        self.cmd_run("oc rsh {} touch /mnt/file{{1..1000}}".format(pod_name))
+        time.sleep(120)
+
+        # Fetching the metrics and storing in initial_metrics as dictionary
+        initial_metrics = self._get_and_manipulate_metric_data(
+            self.metrics, pvc_name)
+        openshift_ops.switch_oc_project(
+            self._master, self.storage_project_name)
+
+        # Get the hostname to reboot where the pod is running
+        pod_info = openshift_ops.oc_get_pods(self._master, name=pod_name)
+        node_for_reboot = pod_info[pod_name]['node']
+
+        # Perform node reboot and wait for ssh connection
+        node_ops.node_reboot_by_command(node_for_reboot)
+
+        # Wait for the ocp node ready and check if any gluster daemons are
+        # running and make sure they are in ready state post node reboot
+        openshift_ops.wait_for_ocp_node_be_ready(self._master, node_for_reboot)
+        if node_for_reboot in self.gluster_servers:
+            gluster_pod = openshift_ops.get_gluster_pod_name_for_specific_node(
+                self._master, node_for_reboot)
+
+            # Wait for glusterfs pod to become ready
+            openshift_ops.wait_for_pod_be_ready(self._master, gluster_pod)
+            services = (
+                ("glusterd", "running"), ("gluster-blockd", "running"),
+                ("tcmu-runner", "running"), ("gluster-block-target", "exited"))
+            for service, state in services:
+                openshift_ops.check_service_status_on_pod(
+                    self._master, gluster_pod, service, "active", state)
+
+        # Create the new pod and validate the prometheus metrics
+        pod_name = openshift_ops.oc_create_tiny_pod_with_volume(
+            self._master, pvc_name, "autotest-volume")
+        self.addCleanup(openshift_ops.oc_delete, self._master, 'pod', pod_name)
+
+        # Wait for POD be up and running and prometheus to refresh the data
+        openshift_ops.wait_for_pod_be_ready(
+            self._master, pod_name, timeout=60, wait_step=2)
+        time.sleep(120)
```

Comment:
Why explcit  wait ?

Reply:
For Prometheus to get the latest info of the pod. It takes approximately 90 seconds in the best case.

Reply:
Same as above

Reply:
I am using the existing method(_get_and_manipulate_metric_data), to change the suggested. I have to change the rest of the other test cases to use the waiter function.

Reply:
ok